[release] v5.11.11 (#36360)
mui/material-ui: CHANGELOG.md

# [Versions](https://mui.com/versions/)

## 5.11.11

<!-- generated comparing v5.11.10..master -->

_Feb 27, 2023_

A big thanks to the 15 contributors who made this release possible. Here are some highlights ✨:

> - 📚 added API documentation for the slots by @hbjORbj, for e.g. [SliderUnstyled API](https://mui.com/base/api/slider-unstyled/#slots)

Reviewer: Only for Base and Joy

- &#8203;<!-- 07 -->[Select] Fix incorrect selecting of first element (#36024) @michaldudak
- &#8203;<!-- 06 -->[Slider] Miscellaneous improvements (#35941) @ZeeshanTamboli
- &#8203;<!-- 05 -->[Slider] Remove unnecessary `data-focusvisible` attribute (#36091) @ZeeshanTamboli
- &#8203;<!-- 04 -->[Snackbar] Replace component logic with `useSnackbar` hook (#36272) @ZeeshanTamboli
- &#8203;<!-- 03 -->[TextField] Fix floating label position (#36246) @oliviertassinari

### `@mui/base@5.0.0-alpha.119`

#### Breaking changes

- &#8203;<!-- 29 -->[base] Remove `classes` prop from the Base components that have it (#36157) @hbjORbj
  These are the components affected by this change: ModalUnstyled, SliderUnstyled, TablePaginationUnstyled and TablePaginationActionsUnstyled. You can replace the `classes` prop by providing the class name prop directly to the prop, for e.g.:

  ```diff
   <TablePaginationUnstyled
  -   classes={{ toolbar: '.toolbar-classname', menuItem: 'menuItem-classname' }}
  +   slotProps={{ toolbar: { className: '.toolbar-classname' }, menuItem: { className: 'menuItem-classname'}}}
   />
  ```

- &#8203;<!-- 28 -->[base] Move hooks to their own directories (#36235) @hbjORbj

>   This is an e.g. of how the migration should look like:

Reviewer: ```suggestion
  This is an example of how the migration should look like:
```

# [Versions](https://mui.com/versions/)

## 5.11.11

<!-- generated comparing v5.11.10..master -->

_Feb 27, 2023_

A big thanks to the 15 contributors who made this release possible. Here are some highlights ✨:

> - 📚 added API documentation for the slots by @hbjORbj, for e.g. [SliderUnstyled API](https://mui.com/base/api/slider-unstyled/#slots)

Author: ```suggestion
- 📚 added API documentation for the slots in MUI Base and Joy UI by @hbjORbj, for e.g. [SliderUnstyled API](https://mui.com/base/api/slider-unstyled/#slots)
```

### Docs

- &#8203;<!-- 27 -->[docs][base] List slots in API documentation (#36104) @hbjORbj
- &#8203;<!-- 26 -->[docs] Add missing sandbox adapter deps resolving (#36291) @LukasTy
- &#8203;<!-- 25 -->[docs] Allow to pass navigation bar banner from outside (#36299) @MBilalShafi
- &#8203;<!-- 24 -->[docs] Fix code on the Working with Tailwind CSS guide (#36090) @mnajdova
- &#8203;<!-- 23 -->[docs] Remove See Slots Section text from Material UI slots description (#36284) @hbjORbj
- &#8203;<!-- 22 -->[docs] Fix emotion warning `:first-child` (#36263) @siriwatknp
- &#8203;<!-- 21 -->[docs][joy] Improve the descriptions of props in API docs (#36307) @hbjORbj
- &#8203;<!-- 20 -->[docs][joy] List slots in API documentation (#36271) @hbjORbj
- &#8203;<!-- 19 -->[docs][joy] Build API documentations (#36008) @hbjORbj
- &#8203;<!-- 18 -->[examples] Update Next.js examples to use built-in font (#36315) @Juneezee
- &#8203;<!-- 17 -->[examples] Update curl link in `material-next-ts-v4-v5-migration` example README (#36321) @ZeeshanTamboli
- &#8203;<!-- 16 -->[examples] Convert Next.js \_document class components to function components (#36109) @ossan-engineer

### Core

> - &#8203;<!-- 08 -->[Rating] Add a comment in Rating component to use `readOnly` state class (#36357) @ZeeshanTamboli

Reviewer 2: This should have been added in '@mui/material' section above by the Changelog script.

Author: This doesn't change anyhow the behavior of the component, why should we bother the developers to read trough it? It's a future message for us, this is why I put it in the Core section.

- &#8203;<!-- 10 -->[Stack][material] Use createStack from the system (#33795) @mnajdova
- &#8203;<!-- 07 -->[Select] Fix incorrect selecting of first element (#36024) @michaldudak
- &#8203;<!-- 06 -->[Slider] Miscellaneous improvements (#35941) @ZeeshanTamboli
- &#8203;<!-- 05 -->[Slider] Remove unnecessary `data-focusvisible` attribute (#36091) @ZeeshanTamboli
- &#8203;<!-- 04 -->[Snackbar] Replace component logic with `useSnackbar` hook (#36272) @ZeeshanTamboli
- &#8203;<!-- 03 -->[TextField] Fix floating label position (#36246) @oliviertassinari
- &#8203;<!-- 13 -->[TextField] Fix floating label position (#36288) @oliviertassinari

### `@mui/base@5.0.0-alpha.119`

#### Breaking changes

- &#8203;<!-- 29 -->[base] Remove `classes` prop from the Base components that have it (#36157) @hbjORbj
  These are the components affected by this change: ModalUnstyled, SliderUnstyled, TablePaginationUnstyled and TablePaginationActionsUnstyled.
  You can replace the `classes` prop by providing the class name prop directly to the prop via `slotProps`. Below is an example of how the migration should look like:

  ```diff
   <TablePaginationUnstyled

>   -   classes={{ toolbar: '.toolbar-classname', menuItem: 'menuItem-classname' }}
>   +   slotProps={{ toolbar: { className: '.toolbar-classname' }, menuItem: { className: 'menuItem-classname'}}}

Reviewer 2: Minor, can be ignored.
```suggestion
  -   classes={{ toolbar: 'toolbar-classname', menuItem: 'menuItem-classname' }}
  +   slotProps={{ toolbar: { className: 'toolbar-classname' }, menuItem: { className: 'menuItem-classname'}}}
```

Author: Let's fix the PR description too, I didn't noticed it :)
